pymeasure/display/widgets/table_widget.py

```diff
@@ -0,0 +1,544 @@
+
+    There are some assumptions:
+    - Series in the dataframe are identical, we call this number k
+    - Series length can be different, we call this number l(x), where x=1..n
+
+    The data can be presented as follow:
+    - By column: each series in a separate column, in this case table shape
+    will be: (k*n) x (max(l(x) x=1..n)
+    - By row: column fixed to the number of series, in this case table shape
+    will be: k x (sum of l(x) x=1..n)
+    """
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.results_list = []
+        self.row_count = 0
+        self.column_count = 0
+
+    def add_results(self, result):
+        if result not in self.results_list:
+            self.results_list.append(result)
+            result.data_changed.connect(partial(self._data_changed, result))
+            self.layoutChanged.emit()
+            result.init()
+            result.start()
+            result.update_data()
+
+    def remove_results(self, result):
+        self.results_list.remove(result)
+        self.row_count = self.pandas_row_count()
+        self.column_count = self.pandas_column_count()
+        result.stop()
+        self.layoutChanged.emit()
+
+    def rowCount(self, parent=None):
+        return self.row_count
+
+    def columnCount(self, parent=None):
+        return self.column_count
+
+    def data(self, index, role=QtCore.Qt.ItemDataRole.DisplayRole):
+        if index.isValid() and role in (QtCore.Qt.ItemDataRole.DisplayRole, SORT_ROLE):
+            result, row, col = self.translate_to_local(index.row(), index.column())
+            try:
+                value = result.data.iloc[row][col]
+                column_type = result.data.dtypes[col]
+                # Cast to column type
+                value_render = column_type.type(value)
+            except IndexError:
+                value = NaN
+                value_render = ""
+            if isinstance(value_render, float64):
+                # limit maximum number of decimal digits displayed
+                value_render = f"{value_render:.{result.float_digits:d}f}"
+
+            if role == QtCore.Qt.ItemDataRole.DisplayRole:
+                return (str(value_render))
+            elif role == SORT_ROLE:
+                # For numerical sort
+                return float(value)
+
+        return None
+
+    def _get_new_rows_columns(self, result, r1, c1, r2, c2):
+        new_rows = self.pandas_row_count() - self.row_count
+        new_rows_start = self.row_count
+
+        new_columns = self.pandas_column_count() - self.column_count
+        new_columns_start = self.column_count
+
+        return new_rows, new_rows_start, new_columns, new_columns_start
+
+    def headerData(self, section, orientation, role):
+        """ Return header information
+
+        Override method from QAbstractTableModel
+        """
+        if role == QtCore.Qt.ItemDataRole.DisplayRole:
+            if orientation == QtCore.Qt.Horizontal:
```

Comment:
I get an error when running a measurement, I think this should state: (also for the if-statements with Qt.Horizontal and Qt.Vertical below)
```suggestion
            if orientation == QtCore.Qt.Orientation.Horizontal:
```

Reply:
Done in commit https://github.com/pymeasure/pymeasure/pull/771/commits/5d400e759c0be400a61724489b6c0b49c03644ae.